editor/src/messages/portfolio/document/node_graph/node_graph_message_handler.rs

```diff
@@ -512,19 +549,27 @@ impl MessageHandler<NodeGraphMessage, (&mut Document, &NodeGraphExecutor, u64, &
 			}
 			NodeGraphMessage::Cut => {
 				responses.add(NodeGraphMessage::Copy);
-				responses.add(NodeGraphMessage::DeleteSelectedNodes);
+				// remove node connections on the cut operation
+				responses.add(NodeGraphMessage::DeleteSelectedNodes { disconnect_input_nodes: None });
 			}
-			NodeGraphMessage::DeleteNode { node_id } => {
+			NodeGraphMessage::DeleteNode { node_id, disconnect_nodes } => {
 				if let Some(network) = self.get_active_network_mut(document) {
-					self.remove_node(network, node_id);
+					self.remove_node(network, node_id, disconnect_nodes);
 				}
 				self.update_selected(document, responses);
 			}
-			NodeGraphMessage::DeleteSelectedNodes => {
+			NodeGraphMessage::DeleteSelectedNodes { disconnect_input_nodes } => {
 				responses.add(DocumentMessage::StartTransaction);
 
+				let disconnect_nodes: bool;
+				if let Some(key) = disconnect_input_nodes {
+					disconnect_nodes = input.keyboard.get(key as usize)
+				} else {
+					disconnect_nodes = false
```

Comment:
This should probably be true?

Reply:
Ah actually we only process the nodes' inputs  if this condition is false. Otherwise in case of a true value, that part is totally skipped during code execution.

Reply:
I didn't think we would reconnect nodes on Ctrl X - although perhaps we should?

Reply:
@Keavon should we reconnect nodes when performing a cut operation?

Reply:
Keavon says it is fine.